<a href="https://colab.research.google.com/github/ainasarfaraz343-a11y/flyrank-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**My Lane: Refresh/ Content Opportunity scoring**
I chose Refresh/ Content Opportunity scoring because the SEO teams cannot update every page at the same time. A ranked list of pages can help them decide which page should be reviewed first. This project focuses on supporting that decision using the available content and performance data instead of relying on guesswork.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision:** Which content pages should be reviewed or refreshed first?
**Action:** An SEO manager or content editor will use the ranked list to decide which pages to update first.
**Cost of Wrong Recommendation:** If the wrong pages are selected, the team may spend time updating pages that do not need attention while missing pages that require urgent updates.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

# Load starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Quick information
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])
print("Clients:", df["client_id"].nunique())

# Supporting numbers
print("Average CTR (%):", round(df["ctr"].mean(), 2))
print("Average Position:", round(df["avg_position"].mean(), 2))
print("Average Content Age (days):", round(df["content_age_days"].mean(), 2))

Rows: 30000
Columns: 44
Clients: 32
Average CTR (%): 0.51
Average Position: 16.34
Average Content Age (days): 256.17


The starter dataset contains 30,000 content pages from 32 clients with 44 features. The average CTR is 0.51%, the average position is 16.34, and the average content age is about 256.17 days. These numbers suggest that there is enough variation in the data to investigate which pages may benefit from a content refresh.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will provide decision support based on observed data. It may identify pages that appear to be good candidates for review, but it cannot prove that refreshing a page will improve Google rankings or traffic. The results will be used as guidance rather than guaranteed predictions.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.